### PR TITLE
Fix issue #224

### PR DESCRIPTION
**Pull request by blasten**

### CODE REVIEW

File: iron-list.html [hunk 1]
```diff
@@ -1033,14 +1033,24 @@
       if (!el || el._templateInstance.__key__ !== key) {
         return;
       }
-      
       if (dot >= 0) {
         path = this.as + '.' + path.substring(dot+1);
         el._templateInstance.notifyPath(path, value, true);
       } else {
+        // Update selection if needed
+        var currentItem = el._templateInstance[this.as];
```
Comment (blasten) [inline iron-list.html]: Keep the selection in sync  e.g.   if `items[0] = 'a';` and `selection[0] = 'a';` if the user does `set('item.0', 'b')`  then  `selection[0]` should be `b`

File: iron-list.html [hunk 2]
```diff
@@ -1033,14 +1033,24 @@
...
+        if (Array.isArray(this.selectedItems)) {
+          for (var i = 0; i < this.selectedItems.length; i++) {
+            if (this.selectedItems[i] === currentItem) {
+              this.set('selectedItems.' + i, value);
+              break;
+            }
+          }
+        } else if (this.selectedItem) {
+          this.set('selectedItem', value);
```
Comment (frankiefu) [inline iron-list.html]: Do we need to check if the selectedItem is the same as the item being modified?

Comment (blasten) [inline iron-list.html]: good catch!

Comment (blasten) [inline iron-list.html]: fixed

File: iron-list.html (hunk 1 above)
Comment (frankiefu) [inline iron-list.html]: Move this inside `if (Array.isArray(this.selectedItems)) {`.

Comment (blasten) [inline iron-list.html]: this is also used by `else if(this.selectedItem === currentItem1)`

Comment (frankiefu) [inline iron-list.html]: You are right.

**Comment by frankiefu** [comment]
LGTM

**Pull request merged**